evm/src/cpu/kernel/asm/bignum/mul.asm

```diff
@@ -12,49 +12,56 @@ global mul_bignum:
     // stack: len, len, a_start_loc, b_start_loc, output_loc, retdest
     ISZERO
     %jumpi(len_zero)
-    DUP1
-    // stack: n=len, len, a_start_loc, bi=b_start_loc, output_cur=output_loc, retdest
+    
+    PUSH @SEGMENT_KERNEL_GENERAL
+    GET_CONTEXT
+    %build_address_no_offset
+
+    DUP2
+    // stack: n=len, base_addr, len, a_start_loc, bi=b_start_loc, output_cur=output_loc, retdest
 mul_loop:
-    // stack: n, len, a_start_loc, bi, output_cur, retdest
+    // stack: n, base_addr, len, a_start_loc, bi, output_cur, retdest
     PUSH mul_addmul_return
-    // stack: mul_addmul_return, n, len, a_start_loc, bi, output_cur, retdest
-    DUP5
-    // stack: bi, mul_addmul_return, n, len, a_start_loc, bi, output_cur, retdest
-    %mload_current_general
-    // stack: b[i], mul_addmul_return, n, len, a_start_loc, bi, output_cur, retdest, b
-    DUP5
-    // stack: a_start_loc, b[i], mul_addmul_return, n, len, a_start_loc, bi, output_cur, retdest, b
-    DUP8
-    // stack: output_loc, a_start_loc, b[i], mul_addmul_return, n, len, a_start_loc, bi, output_cur, retdest, b
+    // stack: mul_addmul_return, n, base_addr, len, a_start_loc, bi, output_cur, retdest
+    DUP6
+    // stack: bi, mul_addmul_return, n, base_addr, len, a_start_loc, bi, output_cur, retdest
+    DUP4 ADD // bi_addr
+    MLOAD_GENERAL
+    // stack: b[i], mul_addmul_return, n, base_addr, len, a_start_loc, bi, output_cur, retdest, b
```

Comment:
Unrelated to the PR, but what's `b` at the end of the stack?

Reply:
no idea, but it's after retdest, so technically unrelevant to this section (probably a typo anyway). I removed it.